secrets/resource_secret.go

```diff
@@ -147,8 +66,16 @@ func ResourceSecret() common.Resource {
 			},
 		},
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			if err := NewSecretsAPI(ctx, c).Create(d.Get("string_value").(string), d.Get("scope").(string),
-				d.Get("key").(string)); err != nil {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			err = w.Secrets.PutSecret(ctx, workspace.PutSecret{
+				Scope:       d.Get("scope").(string),
+				Key:         d.Get("key").(string),
+				StringValue: d.Get("string_value").(string),
+			})
```

Comment:
I think this is fine, but most of our other resources do this as:
```go
var putSecretReq workspace.PutSecret
common.DataToStructPointer(d, s, &putSecretReq)
err = w.Secrets.PutSecret(ctx, putSecretReq)
```
Does this not work?